app/scripts/migrations/077.test.js

```diff
@@ -319,4 +482,824 @@ describe('migration #77', () => {
+            },
+            TokensController: {
+              allTokens: {
+                16: {
+                  '0x111': [
+                    {
+                      address: '0xaaa',
+                      decimals: 1,
+                      symbol: 'TEST1',
+                    },
+                  ],
+                },
+                '0x10': {
+                  '0x111': [
+                    {
+                      address: '0xaaa',
+                      decimals: 1,
+                      symbol: 'TEST1',
+                    },
+                  ],
+                },
+                32: {
+                  '0x222': [
+                    {
+                      address: '0xbbb',
+                      decimals: 1,
+                      symbol: 'TEST2',
+                    },
+                  ],
+                },
+              },
+            },
+          },
+        };
+
+        const newStorage = await migration77.migrate(oldStorage);
+
+        expect(newStorage.data).toStrictEqual({
+          TokenListController: {
+            tokensChainsCache: {},
+          },
+          TokensController: {
+            allTokens: {
+              '0x10': {
+                '0x111': [
+                  {
+                    address: '0xaaa',
+                    decimals: 1,
+                    symbol: 'TEST1',
+                  },
+                ],
+              },
+            },
+          },
+        });
+      });
+
+      it('deletes entries in TokensController.allIgnoredTokens that have decimal chain IDs if there exists an entry with the equivalent chainId in hex', async () => {
+        const oldStorage = {
+          meta: { version: 87 },
+          data: {
+            TokenListController: {
+              tokensChainsCache: {},
+            },
+            TokensController: {
+              allIgnoredTokens: {
+                16: {
+                  '0x1': {
+                    '0x111': ['0xaaa'],
+                  },
+                },
+                '0x10': {
+                  '0x1': {
+                    '0x222': ['0xbbb'],
+                  },
+                },
+                32: {
+                  '0x2': {
+                    '0x222': ['0xbbb'],
+                  },
+                },
+              },
+            },
+          },
+        };
+
+        const newStorage = await migration77.migrate(oldStorage);
+
+        expect(newStorage.data).toStrictEqual({
+          TokenListController: {
+            tokensChainsCache: {},
+          },
+          TokensController: {
+            allIgnoredTokens: {
+              '0x10': {
+                '0x1': {
+                  '0x222': ['0xbbb'],
+                },
+              },
+            },
+          },
+        });
+      });
+
+      it('deletes entries in TokensController.allDetectedTokens that have decimal chain IDs if there exists an entry with the equivalent chainId in hex', async () => {
```

Comment:
Some of these descriptions need updating

```suggestion
      it('deletes entries in TokensController.allDetectedTokens that have decimal chain IDs', async () => {
```